internal/util/importutilv2/parquet/field_reader.go

```diff
@@ -416,6 +425,74 @@ func ReadNullableIntegerOrFloatData[T constraints.Integer | constraints.Float](p
 	return data, validData, nil
 }
 
+// This method returns a []map[string]arrow.Array
+// map[string]arrow.Array represents a struct
+// For example 1:
+//
+//	  struct {
+//		 name string
+//	     age  int
+//	  }
+//
+// The ReadStructData() will return a list like:
+//
+//	  [
+//		 {"name": ["a", "b", "c"], "age": [4, 5, 6]},
+//	     {"name": ["e", "f"], "age": [7, 8]}
+//	  ]
+//
+// Value type of "name" is array.String, value type of "age" is array.Int32
+// The length of the list is equal to the length of chunked.Chunks()
+//
+// For sparse vector, the map[string]arrow.Array is like {"indices": array.List, "values": array.List}
+// For example 2:
+//
+//	  struct {
+//		 indices []uint32
+//	     values  []float32
+//	  }
+//
+// The ReadStructData() will return a list like:
+//
+//	  [
+//		 {"indices": [[1, 2, 3], [4, 5], [6, 7]], "values": [[0.1, 0.2, 0.3], [0.4, 0.5], [0.6, 0.7]]},
+//	     {"indices": [[8], [9, 10]], "values": [[0.8], [0.9, 1.0]]}
+//	  ]
+//
+// Value type of "indices" is array.List, element type is array.Uint32
+// Value type of "values" is array.List, element type is array.Float32
+// The length of the list is equal to the length of chunked.Chunks()
+//
+// Note: now the ReadStructData() is used by SparseVector type and SparseVector is not nullable,
+// create a new method ReadNullableStructData() if we have nullable struct type in future.
+func ReadStructData(pcr *FieldReader, count int64) ([]map[string]arrow.Array, error) {
+	chunked, err := pcr.columnReader.NextBatch(count)
+	if err != nil {
+		return nil, err
+	}
+	data := make([]map[string]arrow.Array, 0, count)
+	for _, chunk := range chunked.Chunks() {
+		structReader, ok := chunk.(*array.Struct)
+		if structReader.NullN() > 0 {
+			return nil, merr.WrapErrParameterInvalidMsg("not nullable, but has null value")
```

Comment:
please help to improve the error msg to "has null value, but struct doesn't support nullable yet"

Reply:
done